web/src/api/utils.ts

```diff
@@ -20,7 +20,8 @@ export const copyToClipboard = async (secret: string) => {
 };
 
 export const makeSharedLinkUrl = (externalDomain: string, key: string) => {
-  return `${externalDomain || window.location.origin}/share/${key}`;
+  const url = externalDomain || window.location.origin;
+  return `${url + url.endsWith('/') ? '' : '/'}share/${key}`;
```

Comment:
Is it possible to use a `URL` constructor instead? I believe this should work (untested):

```suggestion
  const base = externalDomain || window.location.origin;
  return new URL(`/share/${key}`, base).href;
```

Reply:
Oh, that's very well possible. I only did a quick google search (less than 1 minute lol) and people were like "yea, regex" lol
I'll test it real quick

Reply:
Apparently this doesn't work if your domain has a path as well (so `immich.app/public/`. Not sure if that's common though...?

Reply:
That's a good point, though at the moment it's not possible to run Immich under a path anyways.

Reply:
We haven't tested with it and I think there are some things that don't work with it like `goto(/photos)`

Reply:
I know that there are absolute paths but I was thinking you could have proxy rewrite rules for that stuff. Or well, at that point, you could also redirect /share to /public/share, nvm lol